main/core/src/main/java/org/cryptomator/webdav/WebDavServer.java

```diff
@@ -85,7 +91,13 @@ public synchronized void stop() {
 	 */
 	public ServletLifeCycleAdapter createServlet(final Path workDir, final boolean checkFileIntegrity, final Cryptor cryptor) {
 		try {
-			final URI uri = new URI(null, null, localConnector.getHost(), localConnector.getLocalPort(), "/" + UUID.randomUUID().toString(), null, null);
+			/*
+			 * The context path is formed from a UUID and the name of the encrypted folder.
+			 * We'll strip the extension of the name and normalize to a limited character set.
+			 * This gives us usable names on windows.
+			 */
+			final String normalizedName = normalize(StringUtils.removeEndIgnoreCase(workDir.getFileName().toString(), cryptor.getFolderExtension()));
```

Comment:
Add a new param `String vaultName` to the createServlet function rather than decomposing it from workDir and folderExtension. The UI (which calls createServlet) is master for all human readable things.